### PR TITLE
backend: add unit tests for ProductService

### PR DESCRIPTION
**Pull request by ZaytsevGleb**

### CODE REVIEW

File: src/Web/WeightControl.UnitTests/WeightControl.UnitTests.csproj [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <IsPackable>false</IsPackable>
```
Comment (denismaskaliou) [inline src/Web/WeightControl.UnitTests/WeightControl.UnitTests.csproj:5]: Please remove this

File: src/Web/WeightControl.UnitTests/Products/GetProductsServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+            // Assert
+            await Assert.ThrowsAsync<BadRequestException>(() => task);
+
+            mockProductsRepository.Verify(x => x.GetAsync(It.IsAny<int>()), Times.Never);
+            mockMapper.Verify(x => x.Map<ProductDto>(It.IsAny<Product>()), Times.Never);
+        }
+
+        [Theory]
+        [InlineData(1)]
```
Comment (denismaskaliou) [inline src/Web/WeightControl.UnitTests/Products/GetProductsServiceTests.cs]: You don't need Inline data in this case, use just any valid data and Fact attribute

File: src/Web/WeightControl.UnitTests/Products/FindProductsServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        {
+            // Arrange
+            var actualProducts = new List<Product>();
+            var expectedProducts = new List<Product> {
+                new Product {Name = "Cake", Calories = 269, Type = 5, Unit = 1},
+                new Product {Name = "Coffe", Calories = 100, Type = 1, Unit = 0},
+            };
+
+            mocker
+                .GetMock<IRepository<Product>>()
```
Comment (denismaskaliou) [inline src/Web/WeightControl.UnitTests/Products/FindProductsServiceTests.cs:81]: Please format this line

File: src/Web/WeightControl.UnitTests/Products/DeleteProductsServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        private readonly ProductsService produtsService;
+        private readonly AutoMocker mocker;
+
+        public DeleteProductsServiceTests()
+        {
+            mocker = new AutoMocker();
+            produtsService = mocker.CreateInstance<ProductsService>();
+        }
+
```
Comment (denismaskaliou) [inline src/Web/WeightControl.UnitTests/Products/DeleteProductsServiceTests.cs:22]: Please add test for positive case, when product deletes successfully

File: src/Web/WeightControl.UnitTests/Products/DeleteProductsServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            var task = produtsService.DeleteAsync(id);
+
+            // Assert
+            await Assert.ThrowsAsync<BadRequestException>(() => task);
+
+            mocker
+               .GetMock<IRepository<Product>>()
+               .Verify(x => x.GetAsync(It.IsAny<int>()), Times.Never);
+
+            mocker
```
Comment (denismaskaliou) [inline src/Web/WeightControl.UnitTests/Products/DeleteProductsServiceTests.cs]: Are you sure that you use Automapper in Delete method????

File: src/Web/WeightControl.UnitTests/Products/DeleteProductsServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+            // Assert
+            await Assert.ThrowsAsync<NotFoundException>(() => task);
+
+            mocker
+               .GetMock<IRepository<Product>>()
+               .Verify(x => x.GetAsync(It.IsAny<int>()), Times.Once);
+
+            mocker
+                .GetMock<IMapper>()
```
Comment (denismaskaliou) [inline src/Web/WeightControl.UnitTests/Products/DeleteProductsServiceTests.cs]: Are you sure that you use Automapper in Delete method????

File: src/Web/WeightControl.UnitTests/Products/CreateProductsServiceTests.cs [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+            // Act
+            var productDto = await productsService.CreateAsync(inputProductDto);
+
+            // Assert
+            Assert.NotNull(productDto);
+
+            mocker
+                .GetMock<IValidator<ProductDto>>()
+                .Verify(x => x.Validate((It.IsAny<ProductDto>())), Times.Once);
```
Comment (patseshankaevgeniy) [inline src/Web/WeightControl.UnitTests/Products/CreateProductsServiceTests.cs]: Please remove the extra brackets

**Review by patseshankaevgeniy — APPROVED** [review]

**Pull request merged**